diff --git a/node_modules/electron-preferences/index.js b/node_modules/electron-preferences/index.js
index babcbc9..aee6e8d 100644
--- a/node_modules/electron-preferences/index.js
+++ b/node_modules/electron-preferences/index.js
@@ -113,6 +113,10 @@ class ElectronPreferences extends EventEmitter2 {
 
     }
 
+    get browserWindowOverrides() {
+        return this.options.browserWindowOverrides; 
+    }
+
     get preferences() {
 
         return this._preferences;
